internal/project/project_test.go

```diff
@@ -240,6 +240,144 @@ func TestProjectLoad(t *testing.T) {
 		}
 	})
 
+	gitProjectNestedDir := testdata.GitProjectNestedPath()
+
+	t.Run("GitProjectWithNested", func(t *testing.T) {
```

Comment:
If the result of `NewDirProject(gitProjectNestedDir)` and `NewDirProject(gitProjectDir)` is expected to be the same, I suggest simplified test:

```go
pRoot, err := NewDirProject(gitProjectDir)
pNested, err := NewDirProject(gitProjectNestedDir)

// Here, either compare `pRoot.fs` and `p.Nested` or collect load events for both and verify that the contents of slices are equal. The second is more resilient to implementation changes. The first requires a change I suggested in another comment.
```